assets/javascripts/bbcode_color_dialect.js

```diff
@@ -6,6 +6,15 @@
     return text;
   }
 
+  function replaceFontBgColor (text) {
+    while (text != (text = text.replace(/\[bgcolor=([^\]]+)\]((?:(?!\[bgcolor=[^\]]+\]|\[\/bgcolor\])[\S\s])*)\[\/bgcolor\]/ig, function (match, p1, p2, offset, string) {
+      return "<span style='background-color: " + p1 + "'>" + p2 + "</span>";
+    })));
+    return text;
+  }
+
   Discourse.Dialect.addPreProcessor(replaceFontColor);
+  Discourse.Dialect.addPreProcessor(replaceFontBgColor);
   Discourse.Markdown.whiteListTag('font', 'color');
+  Discourse.Markdown.whiteListTag('span', 'style');
```

Comment:
Hmm `style` is kind of a *dangerous* attribute to whitelist...
I would only support whitelisting changes of the `background-color` value.
Can you fix that?